pyaerocom/io/read_eea_aqerep_base.py

```diff
@@ -367,16 +367,40 @@ def read_file(self, filename, var_name, vars_as_series=False):
         # TsType is
         # data_out['var_info'][aerocom_var_name]['ts_type'] = self.TS_TYPE
 
+        # Sometimes the times in the data files are not ordered in time which causes problems when doing
+        # time interpolations later on. Make sure that the data is ordered in time
+        diff_unsorted = (
+            data_dict[self.START_TIME_NAME][1:lineidx]
+            - data_dict[self.START_TIME_NAME][: lineidx - 1]
```

Comment:
I do not really understand how these lines sort the file on time.
Maybe a test to show that it works would be of help

Reply:
Is `lineidx` the last line? If so, this could be written as:

``` python
    diff_unsorted = data_dict[self.START_TIME_NAME][1:] - data_dict[self.START_TIME_NAME][:-1]
```

or as a boolean
``` python
    unsorted = any(data_dict[self.START_TIME_NAME][1:] > data_dict[self.START_TIME_NAME][:-1])
```

Reply:
It seems it could be further simplified with [np.diff]

``` pytyon
    unsorted = (np.diff(data_dict[self.START_TIME_NAME][:lineidx]) > 0).any()
```

[np.diff]: https://numpy.org/doc/stable/reference/generated/numpy.diff.html

Reply:
I did not know np.diff to be honest. You know that I tend not to be very pythonic when it comes specialties / conveniences in Python ;-)
I also wonder if these convenience methods are good for speed or not as they need to test more cases. 
I'll change the code to use np.diff to avoid confusion

Reply:
> I did not know np.diff to be honest.

Me neither, but this seemed to be a problem general enough to have been addressed by numpy.

> I also wonder if these convenience methods are good for speed or not as they need to test more cases.

In general, I'm more concerned abut readability than raw performance.
It does not matter how fast we can arrive to the wrong conclusions because the code does something unexpected.

In any case, I suspect that the real performance hit is on the `np.argsort` call.
BTW, why use `np.argsort` to create an index instead of sorting the data itself?

Reply:
`np.argsort` because I need to reorder the the values according to the time (so the order comes from the time and needs to be applied to the time and the data values). These are pure numpy arrays. 
I have used similar code several times before (in IDL where these vector operations are the only fast code), so I did not find it very complicated ;-), But yes, np.diff if clearer